main.go

```diff
@@ -2,14 +2,59 @@ package main
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
 	"os"
 
 	"github.com/databricks/cli/cmd"
 	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/telemetry"
+	"github.com/databricks/databricks-sdk-go/logger"
 )
 
 func main() {
 	ctx := context.Background()
+
+	// A new process is spawned for uploading telemetry data. We handle this separately
+	// from the rest of the CLI commands.
+	// This is done because [root.Execute] spawns a new process to run the
+	// "telemetry upload" command if there are logs to be uploaded. Having this outside
+	// of [root.Execute] ensures that the telemetry upload process is not spawned
+	// infinitely in a recursive manner.
+	if len(os.Args) == 3 && os.Args[1] == "telemetry" && os.Args[2] == "upload" {
+		var err error
+
+		// By default, this command should not write anything to stdout or stderr.
+		outW := io.Discard
```

Comment:
Note that by default the daemon library redirects both stderr and stdout to dev null.
https://github.com/databricks/cli/blob/4881fd873b6b615c79e5e792cdf62aac86d49642/libs/daemon/daemon.go#L57